### PR TITLE
Add `explain` support

### PR DESCRIPTION
**Pull request by kwong-yw**
Add the ability to run `#explain` on a Dataset. This supports all the output formats that Snowflake offers when using EXPLAIN:

> * JSON: JSON output is easier to store in a table and query.
> * TABULAR: tabular output is generally more human-readable than JSON output.
> * TEXT: formatted text output is generally more human-readable than JSON output.
>
> The default is TABULAR.

### CODE REVIEW

**Review by rafalyesware — APPROVED** [review]
Just one question on the specs, otherwise LGTM!

File: spec/sequel/adapters/snowflake_spec.rb [hunk 1]
```diff
@@ -68,4 +68,36 @@
...
+      db.drop_table(test_table)
+    end
+
+    it "should have explain output" do
+      query = db.fetch("SELECT * FROM #{test_table} WHERE ID=1;")
+
+      expect(query.explain).to be_a_kind_of(String)
+      expect(query.explain(:tabular=>true)).to be_a_kind_of(String)
+      expect(query.explain(:json=>true)).to be_a_kind_of(String)
+      expect(query.explain(:text=>true)).to be_a_kind_of(String)
```
[the remark this answers is not in the document]
Comment (kwong-yw) [reply spec/sequel/adapters/snowflake_spec.rb:100]: This is how the Sequel gem also tests that EXPLAIN is working for [other databases](https://github.com/jeremyevans/sequel/blob/5ff23ef79be2eac05b0f885f57a492eaafb41630/spec/adapters/mysql_spec.rb#L213), and I’m inclined to follow that- while it’s nice to see that we get correct output, it’s kinda an implementation detail on the type of database software and not on the Sequel adapter.

[the remark this answers is not in the document]
Comment (rafalyesware) [reply spec/sequel/adapters/snowflake_spec.rb:100]: That's fair! Sold! 😉

**Review by sebastianfairchild — APPROVED** [review]

**Comment by fakefarm** [comment]
Thank you @kwong-yw! 👏👏

**Pull request merged**